### PR TITLE
Remove legacy localisation script references and actions

### PR DESCRIPTION
**Pull request by iangmaia**
**EDIT**: this PR was re-created in the main repo: https://github.com/wordpress-mobile/release-toolkit/pull/447

## What does it do?
This is a clean up removing the remainder references to the legacy localisation tooling, following up the [#433 PR](https://github.com/wordpress-mobile/release-toolkit/pull/443).

It contains:

- Removal of the actions `ios_localize_project` and `ios_update_metadata`.
- Removal of the methods `localize_project`, `update_metadata` and `strings_files` on `Ios::GitHelper`
- Removal of the parameter `include_metadata` from the method `Ios::GitHelper.commit_version_bump(include_deliverfile:)`

## Related PRs
- This PR is built on top of the [#433 PR](https://github.com/wordpress-mobile/release-toolkit/pull/443)
- This PR contains @mokagio's [draft PR](https://github.com/wordpress-mobile/release-toolkit/pull/370), which can be closed if this is merged.
- Based on the changes done here, [this PR](https://github.com/iangmaia/release-toolkit/pull/3) will remove functionality related to the obsolete `Deliverfile`.

## Checklist before requesting a review

- [x] Run `bundle exec rubocop` to test for code style violations and recommendations
- [x] Add Unit Tests (aka `specs/*_spec.rb`) if applicable
- [x] Run `bundle exec rspec` to run the whole test suite and ensure all your tests pass
- [x] Make sure you added an entry in [the `CHANGELOG.md` file](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/CHANGELOG.md#trunk) to describe your changes under the approprioate existing `###` subsection of the existing `## Trunk` section.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 ### Breaking Changes
 
 - Remove the `skip_glotpress` parameter from the `ios_bump_version_release` action [#443]
+- Remove the actions `ios_localize_project` and `ios_update_metadata`
```
Comment (mokagio) [inline CHANGELOG.md:10]: There's no PR number here, but that's to be expected give this PR is on your fork

File: lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_git_helper.rb [hunk 2]
```diff
@@ -15,7 +15,7 @@ module GitHelper
         # @param [Bool] include_deliverfile If true (the default), includes the `fastlane/Deliverfile` in files being commited
         #
         def self.commit_version_bump(include_deliverfile: true)
-          files_list = [File.join(ENV['PROJECT_ROOT_FOLDER'], 'config', '.')]
+          files_list = [File.join(get_from_env!(key: 'PROJECT_ROOT_FOLDER'), 'config', '.')]
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_git_helper.rb:18]: Nice that you updated this while touching the file, and that it was done in a dedicated commit 👍

**Review by mokagio — APPROVED** [review]
Looking good. 

I checked this locally and run the test, which passed. 

I didn't go down the route of opening a PR for this on the main repo to trigger CI because I see this is incremental on top of another one and I'd like to know how you plan to merge them.

**Comment by iangmaia** [comment]
> Looking good.
> 
> I checked this locally and run the test, which passed.
> 
> I didn't go down the route of opening a PR for this on the main repo to trigger CI because I see this is incremental on top of another one and I'd like to know how you plan to merge them.

Thanks @mokagio 👍 

My plan is: once https://github.com/wordpress-mobile/release-toolkit/pull/443 gets merged, I'll target this PR to the main repo's trunk, but pleas let me know if you have other ideas or if another process is preferred.

**Comment by AliSoftware** [comment]
Just merged wordpress-mobile#443 🙂

**Comment by iangmaia** [comment]
Ah, just noticed that you can't switch forks when editing the Pull Request (which makes sense, the PRs are for a repo), only branches... Anyway, I'll just create another one for this branch to the main repo now that 443 has been merged 🙂
